src/Logging/Logging/src/LoggerFactory.cs

```diff
@@ -206,5 +219,34 @@ private struct ProviderRegistration
             public ILoggerProvider Provider;
             public bool ShouldDispose;
         }
+
+        private class DisposingLoggerFactory: ILoggerFactory
```

Comment:
The fact that we need to hold onto the container means all of the sample code is sorta busted

Reply:
Busted how?

Reply:
Maybe you’re saying we don’t need to dispose the service provider?

Reply:
Why does this wrapper exist to begin with?

Reply:
To dispose ILoggerProviders created by DI

Reply:
Doesn’t disposing the factory do that?

Reply:
Ah, I see it wouldn’t. That makes the current usage without this API leaky.

Reply:
If you don't dispose provider - yes.

Reply:
I’m taking about examples that you would use today (even in our samples)

Reply:
Pushed a sample fix.